File: 3kurs/2sem/paralelki/stsala/lab2/src/main/scala/Cpu.scala
import scala.util.Random

class Cpu(mainQueue: CpuQueue,
          secondaryQueue: CpuQueue,
          baseProcessingTime: Int,
          name: String) extends Thread(name) {
  lazy private val r = Random
  
  override def run(): Unit = {
    while (true) {
      val processingTime = r.between(10, 50) + baseProcessingTime
      val res = mainQueue.pop(secondaryQueue)
      println(s"cpu ${name} processing ${res}")
      Thread.sleep(processingTime)
      println(s"cpu ${name} processed ${res} in time ${processingTime}")
    }
    
  }
}

object Cpu {
  def apply(mainQueue: CpuQueue,
            secondaryQueue: CpuQueue,
            baseProcessingTime: Int,
            name: String): Cpu = new Cpu(mainQueue, secondaryQueue, baseProcessingTime, name)
}

File: 3kurs/2sem/paralelki/jaba/lab2/src/main/java/Cpu.java
public class Cpu implements Runnable {
    
    CpuQueue queue;
    int index;
    
    Cpu(CpuQueue q, int index) {
        this.queue = q;
        this.index = index;
    }
    
    public void run() {
        long processingTime;
        while (true) {
            int randMin = 20;
            int randMax = 80; // rand = [20,100]
            processingTime = randMin + (int) (Math.random() * randMax);
            int res = -1;
            try {
                res = queue.get();
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
            System.out.println("CPU " + index + " processed " + res + " in time " + processingTime + "\n");
            try {
                Thread.sleep(processingTime);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }
}
